### DISCUSSION TITLE
Exclude certain IPs from RBL check instead of disabling RBL?

### DISCUSSION THREAD

**Discussion by fernandomm** [rspamd/rspamd#3649] [Q&A]
I have the following RBL configuration:

```
rbls {
    rbl_custom {
        ignore_defaults = true;
        resolve_ip = true;
        checks ['urls', 'content_urls'];
        images = true;
        symbol = "RBL_CUSTOM";
        rbl = "custom-rbl.net";
    }
}
```

It resolves the IP for each url and then check against the RBL. I would like to exclude certain resolved IPs from this check but would like to perform it against other IPs. To be more clear, if I have the following URLs + resolved IPs:

```
https://example1.com -> 1.1.1.1
https://example2.com -> 2.2.2.2
https://example3.com -> 3.3.3.3
```

I would like to exclude 2.2.2.2 and perform the RBL check only for 1.1.1.1 and 3.3.3.3.

There is local_exclude_ip_map but it seems to be used to enable/disable the RBL based on the message origin IP, which is different from what i want.

**Comment by fatalbanana** [comment]
> There is local_exclude_ip_map but it seems to be used to enable/disable the RBL based on the message origin IP, which is different from what i want.

I think it'd be reasonable that we check resolved IPs in this map too; it's true for received header checks so far.

We might want to consider adding a similar facility for things other than IP addresses.
